Sync dcos-launch with latest acs-engine changes

ionutbalutoiu opened the pull request:
This pull requests adds the following updates:

* Set `orchestratorRelease` instead of `orchestratorVersion`
* Add config option for acs-engine `orchestratorRelease`
* Fix capitalization for acs-engine template `osType`
* Remove hard-coded `agentWindowsOffer` and `agentWindowsSku`
* Add acs-engine configs: `windows_publisher`, `windows_offer` and `windows_sku`

ibalutoiu-cbsl committed:
Use orchestratorRelease instead of orchestratorVersion

Setting orchestratorRelease instead of orchestratorVersion will
make acs-engine use the latest DC/OS version for the major release
specified.

cprovencher reviewed (changes requested):
Ideally we should be able to pass this through the config

ionutbalutoiu commented:
@cprovencher, I'll make the change.

ibalutoiu-cbsl committed:
Make acs-engine orchestratorRelease a config option

ionutbalutoiu commented:
@cprovencher, please see my updated PR with the requested change.

ibalutoiu-cbsl committed:
Remove hard-coded agentWindowsOffer and agentWindowsSku

ibalutoiu-cbsl committed:
Fix capitalization for acs-engine template osType

From now on, acs-engine is case-sensitive and will reject any osType
different than: 'Linux' or 'Windows'

ibalutoiu-cbsl committed:
Add acs-engine configs: windows_publisher, windows_offer and windows_sku

ibalutoiu-cbsl committed:
Fix flake8 error
